cpp/src/gandiva/encrypt_utils.cc

```diff
@@ -16,22 +16,38 @@
 // under the License.
 
 #include "gandiva/encrypt_utils.h"
+#include <string.h>
 
 #include <stdexcept>
 
 namespace gandiva {
 GANDIVA_EXPORT
-int32_t aes_encrypt(const char* plaintext, int32_t plaintext_len, const char* key,
-                    unsigned char* cipher) {
+int32_t aes_encrypt(const char* plaintext, int32_t plaintext_len, const char* key, 
+                    int32_t key_len, unsigned char* cipher) {
   int32_t cipher_len = 0;
   int32_t len = 0;
   EVP_CIPHER_CTX* en_ctx = EVP_CIPHER_CTX_new();
+  const EVP_CIPHER* cipher_algo = nullptr;
 
   if (!en_ctx) {
     throw std::runtime_error("could not create a new evp cipher ctx for encryption");
   }
 
-  if (!EVP_EncryptInit_ex(en_ctx, EVP_aes_128_ecb(), nullptr,
+  switch (key_len) {
```

Comment:
This could be combined with the other similar code below into a helper function.